shards/rust/src/types.rs

```diff
@@ -319,6 +320,19 @@ impl Drop for Wire {
   }
 }
 
+// typedef void(__cdecl *SHSetWireError)(const SHWire *, void *errorData, struct SHStringWithLen msg);
+type SHSetWireError = unsafe extern "C" fn(*const SHWire, *mut c_void, SHStringWithLen);
+
+// void shards_set_wire_error_callback(SHWire *wire, SHSetWireError setWireError, void *errorData)
+// the above is defined c side, lets declare it here
+extern "C" {
```

Comment:
This prevents using the rust library when building dll shards, not sure if you care about that still?

Reply:
Would have to place it in SHCore to be correct

Reply:
I think I fixed those by using shards-langffi

Reply:
see I don't want it in SHCore cos it's rather internal.

Reply:
I will test it though